crates/astria-conductor/src/conductor/inner.rs

```diff
@@ -44,14 +48,28 @@ impl std::fmt::Display for RestartOrShutdown {
     }
 }
 
-struct ShutdownSignalReceived;
+pub(crate) enum ExitReason {
+    ShutdownSignalReceived,
+    ChannelsClosed(Box<State>),
```

Comment:
This is an implementation detail. It's not clear why an exit reason `channels is closed` carries a "state" value.